### PR TITLE
Fixed intersphinx bug and formatted docs/conf.py

### PR DESCRIPTION
**Pull request by bdpedigo**
- [x] Does this PR have a descriptive title that could go in our release notes?
- [ ] Does this PR add any new dependencies?
- [ ] Does this PR modify any existing APIs?
   - [ ] Is the change to the API backwards compatible?
- [x] Have you built the documentation (reference and/or tutorial) and verified the generated documentation is appropriate?

#### Reference Issues/PRs
Fixes an issue with docs building on many current open PRs 

#### What does this implement/fix? Briefly explain your changes.
A default option changed in intersphinx which broke one of our matplotlib references

#### Any other comments?
I also ran black on this file accidentally but I figured that we might as well have it match formatting for the rest of the codebase

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -88,10 +88,18 @@
     "sklearn": ("https://scikit-learn.org/dev", None),
 }
 
+intersphinx_disabled_reftypes = []
```
Comment (bdpedigo) [inline docs/conf.py:91]: this is the only substantive change in this PR; the rest is from `black`

**Comment by bdpedigo** [comment]
hmmm this did fix the error locally for me so I'm not sure what's going on. 

see docs here https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html#confval-intersphinx_disabled_reftypes

will revisit tomorrow unless anyone has any ideas

**Comment by bdpedigo** [comment]
@daxpryce this looks to be working - don't ask me why but the `matplotlib:`s appeared to be related to the issue. 

I am a firm believer in "if sphinx aint broke dont try to understand how it works," if you agree, I'd like to merge so I can get this out of the way for a few PRs in the pipeline

**Comment by daxpryce** [comment]
> @daxpryce this looks to be working - don't ask me why but the `matplotlib:`s appeared to be related to the issue.
> 
> I am a firm believer in "if sphinx aint broke dont try to understand how it works," if you agree, I'd like to merge so I can get this out of the way for a few PRs in the pipeline

I agree; I only added the matplotlib because that fixed it when it was broken for the last month, and now it breaks it again?  I should have been a hobbit gardener.

**Review by daxpryce — APPROVED** [review]

**Comment by bdpedigo** [comment]
thanks @daxpryce!

**Pull request merged**